projects/button_challenge/src/main.c

```diff
@@ -0,0 +1,64 @@
+#include <stdint.h>
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "wait.h"
+
+// Callback function
```

Comment:
Though these comments are okay for the firmware homework. We normally wouldn't put such comments since the name of the function or variable itself is usually self-explanatory (eg we know main() is Main). 

You can leave these though since this is just the homework.